ord_schema/validations.py

```diff
@@ -215,95 +201,48 @@ def validate_reaction_input(message):
-    # identifiers until we can resolve one to a SMILES string
-    if not any(identifier.type in _COMPOUND_STRUCTURAL_IDENTIFIERS for
-               identifier in message.identifiers):
-        for identifier in message.identifiers:
-            if identifier.type == identifier.NAME:
-                try:
-                    smiles = pubchem_resolve('name', identifier.value)
-                    new_identifier = message.identifiers.add()
-                    new_identifier.type = new_identifier.SMILES
-                    new_identifier.value = smiles
-                    new_identifier.details = 'NAME resolved by PubChem'
-                    break
-                except urllib.error.HTTPError:
-                    pass
-
-    # Validate compound identifiers that can be parsed by RDKit. At the same
-    # time, create an RDKIT_BINARY identifier for the first valid structure
-    # that we are able to parse.
-    # TODO(ccoley) add more sources for RDKIT_BINARY
-    if Chem and not any(identifier.type == identifier.RDKIT_BINARY for
-                        identifier in message.identifiers):
-        mol = None
-        for identifier in message.identifiers:
-            this_mol = None
-
-            if Chem and identifier.type == identifier.SMILES:
-                this_mol = Chem.MolFromSmiles(identifier.value)
-                if this_mol is None:
-                    warnings.warn(f'RDKit {RDKIT_VERSION} could not validate'
-                                  f' SMILES identifier {identifier.value}',
-                                  ValidationError)
-            elif identifier.type == identifier.INCHI:
-                this_mol = Chem.MolFromInchi(identifier.value)
-                if this_mol is None:
-                    warnings.warn(f'RDKit {RDKIT_VERSION} could not validate'
-                                  f' InChI identifier {identifier.value}',
-                                  ValidationError)
-            elif identifier.type == identifier.MOLBLOCK:
-                this_mol = Chem.MolFromMolBlock(identifier.value)
-                if this_mol is None:
-                    warnings.warn(f'RDKit {RDKIT_VERSION} could not validate'
-                                  ' MolBlock identifier', ValidationError)
-
-            # If RDKit mol was not defined, define it with this resolved mol
-            mol = mol or this_mol
-
-        if mol is not None:
-            # TODO(ccoley) We will want to add more canonicalization,
-            # sanitization, etc., and separate this into its own function.
-            new_identifier = message.identifiers.add()
-            new_identifier.type = new_identifier.RDKIT_BINARY
-            new_identifier.bytes_value = mol.ToBinary()
-    # pylint: enable=too-many-branches
-
-    return message
+    if not Chem or any(identifier.type == identifier.RDKIT_BINARY
+                       for identifier in message.identifiers):
+        return
```

Comment:
- If `Chem` is `None`, we should probably issue a `ValidationWarning` at the very least to indicate that it wasn't possible to validate the compound identifiers.
- If `identifier.type == identifier.RDKIT_BINARY`, I don't think we should assume we're in the clear. Maybe try to initialize a `Chem.Mol` and  `if mol is None: warnings.warn(..., ValidationError)`

Reply:
Also, what if the only identifier is a `NAME`? Should we issue a `ValidationWarning` here? It might be worth mentioning at this validation stage that we haven't yet tried to resolve the name.

Reply:
* I think we should require rdkit for these; the conditional imports don't really make sense since we'll want to run all the validations.
* Agreed that RDKIT_BINARY isn't enough.
* Yes, I think every reaction should have at least one structural identifier, but I wanted to check with you since I remembered that some Compounds might only have NAMEs b/c they have no defined structure...a warning at the Compound level is a good idea.

Reply:
The warning for compounds w/o structural identifier is perfect, thanks for adding that